app/models/claim.rb

```diff
@@ -30,17 +30,17 @@ class Claim < ApplicationRecord
   belongs_to :provider
   belongs_to :created_by, polymorphic: true
 
-  has_many :mentor_trainings
+  has_many :mentor_trainings, dependent: :destroy
   has_many :mentors, through: :mentor_trainings
 
   validates :status, presence: true
   validates :reference, uniqueness: { case_sensitive: false }, allow_nil: true
 
-  scope :not_internal, -> { where.not(status: :internal) }
+  scope :not_internal_or_discarded, -> { where.not(status: %i[internal discarded]) }
```

Comment:
Might be better to have an `active` scope, rather than this very specifically named scope.
Could be beneficially to have a constant (e.g `ACTIVE_STATUSES = %i[draft submitted].freeze`), rather than doing the negative of that search. 
```suggestion
  scope :active, -> { where(status: ACTIVE_STATUSES) }
```